apps/api/src/coscrad-cli/discover-audio-items.cli-command.ts

```diff
@@ -0,0 +1,90 @@
+import { LanguageCode } from '@coscrad/api-interfaces';
+import { isBoolean } from '@coscrad/validation-constraints';
+import { writeFileSync } from 'fs';
+import { TermQueryService } from '../domain/services/query-services/term-query.service';
+import { InternalError } from '../lib/errors/InternalError';
+import { CliCommand, CliCommandOption, CliCommandRunner } from './cli-command.decorator';
+
+interface DiscoverAudioItemsCliCommandOptions {
+    filepath: string;
+    languageCode: LanguageCode;
+    publish: boolean;
+}
+
+@CliCommand({
+    name: 'discover-audio-for-terms',
+    description: 'export a json file identifying possible audio items for each term without audio',
+})
+export class DiscoverAudioItemsCliCommand extends CliCommandRunner {
+    constructor(private readonly termQueryService: TermQueryService) {
+        super();
+    }
+
+    async run(
+        _passedParams: string[],
+        {
+            filepath,
+            languageCode: languageCodeForAudio,
+            publish: shouldPublishTerms,
+        }: DiscoverAudioItemsCliCommandOptions
+    ): Promise<void> {
+        const audioForTerms = await this.termQueryService.discoverAudio({
+            languageCodeForAudio,
+            // TODO CLI optoin
+            shouldPublishTerms,
+        });
+
+        // TODO add time stamp
+        writeFileSync(
+            filepath,
+
+            JSON.stringify(audioForTerms, null, 4)
+        );
+    }
+
+    @CliCommandOption({
+        flags: '-f, --filepath [filepath]',
+        description: 'the path to the desired output file',
+        required: true,
+    })
+    parseFilepath(value: string): string {
+        // if(existsSync(value)){
```

Comment:
remove commented code?

Reply:
Actually this is a problem. I need to add test coverage and implement this.

Reply:
I've fixed this now.